### ISSUE TITLE
Conflict between SpatialImage class name from SpatialExperiment and Seurat

### ISSUE THREAD

**Issue by lmweber** [drighelli/SpatialExperiment#98]
Currently there is an issue where users are getting errors if both SpatialExperiment and Seurat are installed at the same time.

This has been discussed previously in the Seurat repo at https://github.com/satijalab/seurat/issues/5128 and came up again in discussions today.

I previously thought this was only an issue with older versions of SpatialExperiment, but seems this is not the case.

I think the issue is that both SpatialExperiment and Seurat have a class called `SpatialImage`, so if users have both installed and are trying to run Seurat code, then Seurat is using the wrong `SpatialImage` class and returns an error.

This is unfortunately a problem for us since it seems the simplest solution / advice given for Seurat users is simply to uninstall SpatialExperiment. So I am worried people are simply going to stop using SpatialExperiment (and our related packages) if we can't find a way to fix this. It is just an unfortunate coincidence that we both picked the same class name.

Possible ideas for solutions:
- put `::` everywhere, but then this only solves the problem in one direction (when using SpatialExperiment, not when using Seurat)
- rename our `SpatialImage` class, which would be quite a hassle

**Comment by lmweber** [comment]
Probably right now what is happening in people's analysis scripts is that R is choosing which `SpatialImage` class to use based on which order the two classes are loaded, i.e.

```
library(SpatialExperiment)
library(Seurat)
```

vs.

```
library(Seurat)
library(SpatialExperiment)
```

So this is also why it is quite a cryptic error and only happen sometimes and not other times.

**Comment by lmweber** [comment]
If we do need to rename `SpatialImage` then I would suggest `SpatialImageSPE`. But maybe there is some better way to solve which class R chooses.

**Comment by lmweber** [comment]
In this case we should also check if we have any conflicts for any other class names (although I think we only have SPE class and the SpatialImage classes, will double check)

**Comment by lmweber** [comment]
Some ideas for class names: `SpatialImageSPE`, `SPEImage`, `SPEimage`

**Comment by LTLA** [comment]
Note that `SpatialImage` itself is a virtual class so it can pretty much be called anything. Just stick an `Abstract` in front of it like Julia would do, i.e., `AbstractSpatialImage`. No one should be/can explicitly create an instance of an SI anyway.

**Comment by lmweber** [comment]
Ah interesting, thanks for this idea

**Comment by PeteHaitch** [comment]
FWIW I think there's another similar-ish instance of this with the `Assays` class defined in **SummarizedExperiment** and the `Assay` class, with the `Assays` function, defined in **Seurat**/**SeuratObject**.

**Comment by lmweber** [comment]
This has now been addressed in PR #101 . We renamed the `SpatialImage` class to `VirtualSpatialImage`, so there is no longer a conflict with Seurat.
